pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -197,6 +197,9 @@ def on_advance_end(self):
             # progress global step according to grads progress
             self.global_step += 1
 
+        # if fault tolerant is enabled and process has been notified, exit.
+        self.trainer.should_exit_gracefully()
```

Comment:
So the plan is to add this everywhere we can, right?

Should we do one PR for each occurrence with its test?

Reply:
Right now, I am adding it only for batch level (train, val loops), so let me add 2 tests.

Reply:
We would do another PR for epoch end if required.

Reply:
Let's use this PR to PoC it and split it afterward.